connect/runtime/src/test/java/org/apache/kafka/connect/integration/InternalTopicsIntegrationTest.java

```diff
@@ -135,11 +138,18 @@ public void testFailToCreateInternalTopicsWithMoreReplicasThanBrokers() throws I
         log.info("Completed startup of {} Kafka broker. Expected Connect worker to fail", numBrokers);
 
         // Try to start a worker
-        connect.addWorker();
+        try (LogCaptureAppender logCaptureAppender = LogCaptureAppender.createAndRegister(DistributedHerder.class)) {
+            connect.addWorker();
+            TestUtils.waitForCondition(() -> logCaptureAppender.getEvents().stream().anyMatch(e -> e.getLevel().equals("ERROR")) &&
+                    logCaptureAppender.getEvents().stream().anyMatch(e ->
+                            e.getMessage().contains("Uncaught exception in herder work thread, exiting:")
+                            ), 60000, "internal topic creation failed");
```

Comment:
`worker.start()` attempt 6s, after timeout will throw exception.

Reply:
Looks like 60 seconds instead of 6, right? Which is probably fine (our CI can be _very_ slow sometimes).

Still, I don't love that we're using log messages to verify that the herder has died. It's brittle, and also prevents us from verifying the actual cause of death.

Could we do something like tracking the `Future<?>` created when we start the herder thread [here](https://github.com/apache/kafka/blob/206d0f809ad8438e9d89b7f731a8b995f952f0c0/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java#L366), then invoking `get` on it (probably with a timeout), and verifying that that call throws an `ExecutionException`?

Or, if for some reason that's difficult, we could also use the existing [herderThread field](https://github.com/apache/kafka/blob/206d0f809ad8438e9d89b7f731a8b995f952f0c0/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java#L231) and invoke `join` (again, probably with a timeout) on it to wait for the herder thread to terminate.